website/docs/advanced-concepts/access-control/granular-access-control/roles.md

```diff
@@ -0,0 +1,242 @@
+
+Delete permission behaves differently at various hierarchical levels -
+
+##### Workspace 
+
+If you give delete permission to the workspace, It gives access to remove any application from the respective workspace. 
+
+##### Application
+
+If you want the user to have delete access only to a specific application in a workspace, open the respective workspace's tree and check the delete permission adjacent to the respective application. **Delete** for just an application allows you to delete the pages of that application.
+
+##### Page
+
+**Delete** access for a particular page allows you to delete the respective page or the resources in the page like APIs, queries, JSObjects. 
+
+#### Export
+
+**Export** allows a user to export the applications in a workspace. If a user has export permission enabled, he can also view the applications in the workspace.
+This permission can also be enabled for a particular application, allowing the user to export and view the respective application only.
+
+#### Make public
+
+**Make Public** gives the user an ability to make the application public or private in a workspace.
+
+### Datasource and queries permissions
+
+For datasource and queries, here are the permissions that can be assigned to user or a group -
+
+
+#### Execute
+
+**Execute** gives a user permission to run DB/API queries created on a datasource. This permission can be given for a workspace or for an individual datasource in a workspace.
+
+##### Workspace
+
+Giving execute permission at a workspace level allows the user to execute any datasource in the workspace.
+
+##### Datasource
+
+If you want the user to have execute access only to a specific datasource in a workspace, open the respective workspace tree and check the execute permission adjacent to the respective datasource. **Execute** for a specific datasource allows the user to execute only that datasource.
+
+##### Page
+
+Going further down the hierarchy, you can give execute access to the queries inside a page. Execute permission at this stage allows the user to run the respective query. Execute permission for a query in page only works if the corresponding datasource also has the execute access.
+
+#### Create
+
+**Create** gives the ability to create datasources in a workspace. In Datasource and queries, you can give the create permission at different hierarchical levels -
+
+##### Workspace
+
+If you give create permission to the workspace, it gives access to create datasources in the respective workspace. 
+
+##### Datasource
+
+If you want the user to have create access only to a specific datasource in a workspace, open the respective workspace tree and check the create permission adjacent to the respective datasource. **Create** for a particular datasource allows the user to make tweaks the datasource configurations and further test, delete or save it.
```

Comment:
I don't think this is right:
`**Create** for a particular datasource allows the user to make tweaks the datasource configurations and further test, delete or save it.` 

Create on a particular datasource allows you to create queries on that datasource but you also need the access on Application resources tab as mentioned below by you.

CC: @trishaanand

Reply:
Yes, but we are specifically talking about create access to datasource here. A user can create queries if he has create access to datasource and create access for the queries.

Reply:
yes but allows the user to make tweaks gives an impression that they have the edit access to datasource. Its giving access to create queries on top of that datasource which is not clear in this statement.